flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvToRowDataConverters.java

```diff
@@ -305,7 +311,7 @@ private CsvToRowDataConverter createArrayConverter(ArrayType arrayType) {
     }
 
     private static void validateArity(int expected, int actual, boolean ignoreParseErrors) {
-        if (expected != actual && !ignoreParseErrors) {
+        if (expected > actual && !ignoreParseErrors) {
```

Comment:
Can you double-check if this is still needed?

Reply:
The `CsvFormat.Reader` read() method needs to read elements according to the full schema from the file but emmit elements with a potentially reduced number of rows according to the projection pushdown.